simulator/src/main/java/com/hedera/block/simulator/mode/PublisherModeHandler.java

```diff
@@ -170,7 +181,9 @@ private void constantRateStreaming() throws InterruptedException, IOException, B
      * Stops the handler and manager from streaming.
      */
     @Override
-    public void stop() {
+    public void stop() throws InterruptedException {
         shouldPublish.set(false);
+        publishStreamGrpcClient.completeStreaming();
+        publishStreamGrpcClient.shutdown();
```

Comment:
See comment from `ConsumerModeHandler` for the same teardown logic.

Reply:
Fixed.